## `nixpkgs-review` result

Generated using [`nixpkgs-review-gha`](https://github.com/Defelo/nixpkgs-review-gha)

Command: `nixpkgs-review pr 427977`

Logs: https://github.com/Prince213/nixpkgs-review-gha/actions/runs/16503928622


---
### `x86_64-darwin` (sandbox = relaxed)
<details>
  <summary>:x: 2 packages failed to build:</summary>
  <ul>
    <li>fava</li>
    <li>fava.dist</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>beanquery (python313Packages.beanquery)</li>
    <li>beanquery.dist (python313Packages.beanquery.dist)</li>
    <li>python312Packages.beanquery</li>
    <li>python312Packages.beanquery.dist</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `x86_64-darwin`</summary>
<details>
<summary>fava</summary>
<pre>        with watcher:
            watcher.update([watcher_paths.file1], [])
    
            watcher_paths.file1.unlink()
>           assert _watcher_poll_check(watcher)
E           assert False
E            +  where False = _watcher_poll_check(<fava.core.watcher.WatchfilesWatcher object at 0x1138c4e10>)

tests/test_core_watcher.py:133: AssertionError
=============================== warnings summary ===============================
tests/test_cli.py:69
  /nix/build/nix-build-fava-1.30.5.drv-4099-4114062188/fava-frontend-1.30.5/tests/test_cli.py:69: PytestUnknownMarkWarning: Unknown pytest.mark.no_cover - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    @pytest.mark.no_cover

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
=========================== short test summary info ============================
FAILED tests/test_core_watcher.py::test_watchfiles_watcher - assert False
FAILED tests/test_core_watcher.py::test_watchfiles_watcher_recognises_change_to_previously_deleted_file - assert False
ERROR tests/test_cli.py::test_cli - PermissionError: [Errno 1] Operation not permitted
======== 2 failed, 452 passed, 2 skipped, 1 warning, 1 error in 10.12s =========</pre>
</details>
</details>


---
### `aarch64-darwin` (sandbox = relaxed)
<details>
  <summary>:x: 2 packages failed to build:</summary>
  <ul>
    <li>fava</li>
    <li>fava.dist</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>beanquery (python313Packages.beanquery)</li>
    <li>beanquery.dist (python313Packages.beanquery.dist)</li>
    <li>python312Packages.beanquery</li>
    <li>python312Packages.beanquery.dist</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `aarch64-darwin`</summary>
<details>
<summary>fava</summary>
<pre>        with watcher:
            watcher.update([watcher_paths.file1], [])
    
            watcher_paths.file1.unlink()
>           assert _watcher_poll_check(watcher)
E           assert False
E            +  where False = _watcher_poll_check(<fava.core.watcher.WatchfilesWatcher object at 0x1073dce10>)

tests/test_core_watcher.py:133: AssertionError
=============================== warnings summary ===============================
tests/test_cli.py:69
  /nix/build/nix-build-fava-1.30.5.drv-3804-2841852365/fava-frontend-1.30.5/tests/test_cli.py:69: PytestUnknownMarkWarning: Unknown pytest.mark.no_cover - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    @pytest.mark.no_cover

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
=========================== short test summary info ============================
FAILED tests/test_core_watcher.py::test_watchfiles_watcher - assert False
FAILED tests/test_core_watcher.py::test_watchfiles_watcher_recognises_change_to_previously_deleted_file - assert False
ERROR tests/test_cli.py::test_cli - PermissionError: [Errno 1] Operation not permitted
========= 2 failed, 452 passed, 2 skipped, 1 warning, 1 error in 6.94s =========</pre>
</details>
</details>

--- a/pkgs/by-name/fa/fava/dont-compile-frontend.patch
+++ b/pkgs/by-name/fa/fava/dont-compile-frontend.patch
@@ -1,0 +1,11 @@
+--- a/_build_backend.py
++++ b/_build_backend.py
+@@ -87,7 +87,7 @@
+ 
+ def _build_fava() -> None:
+     """Run the build steps for Fava."""
+-    _compile_frontend()
++    # _compile_frontend()
+     _compile_translations()
+ 
+ 

--- a/pkgs/by-name/fa/fava/package.nix
+++ b/pkgs/by-name/fa/fava/package.nix
@@ -90,6 +90,7 @@ python3Packages.buildPythonApplication {
     license = lib.licenses.mit;
     maintainers = with lib.maintainers; [
       bhipple
+      prince213
       sigmanificient
     ];
   };

--- a/pkgs/by-name/fa/fava/package.nix
+++ b/pkgs/by-name/fa/fava/package.nix
@@ -2,16 +2,17 @@
   lib,
   python3Packages,
   fetchPypi,
+  stdenv,
 }:
 
 python3Packages.buildPythonApplication rec {
   pname = "fava";
-  version = "1.30.1";
+  version = "1.30.5";
   pyproject = true;
 
   src = fetchPypi {
     inherit pname version;
-    hash = "sha256-69Wx9/H7nLDPZP9LOUnDJngY9YTCcr+oQ0E6+xeIWPE=";
+    hash = "sha256-pfnRNhAcyuYFHqPBF0qCrK7w1PJiMOdYXCGj+xXi6uQ=";
   };
 
   postPatch = ''
@@ -39,6 +40,12 @@ python3Packages.buildPythonApplication rec {
   ];
 
   nativeCheckInputs = [ python3Packages.pytestCheckHook ];
+
+  # tests/test_cli.py
+  __darwinAllowLocalNetworking = true;
+
+  # flaky, fails only on ci
+  disabledTestPaths = lib.optionals stdenv.hostPlatform.isDarwin [ "tests/test_core_watcher.py" ];
 
   env = {
     # Disable some tests when building with beancount2

--- a/pkgs/by-name/fa/fava/package.nix
+++ b/pkgs/by-name/fa/fava/package.nix
@@ -1,19 +1,45 @@
 {
   lib,
   python3Packages,
-  fetchPypi,
+  buildNpmPackage,
+  fetchFromGitHub,
   stdenv,
 }:
+let
+  src = buildNpmPackage (finalAttrs: {
+    pname = "fava-frontend";
+    version = "1.30.5";
 
-python3Packages.buildPythonApplication rec {
+    src = fetchFromGitHub {
+      owner = "beancount";
+      repo = "fava";
+      tag = "v${finalAttrs.version}";
+      hash = "sha256-46ze+1sdgXq9Unhu1ec4buXbH3s/PCcfCx+rmYc+fZw=";
+    };
+    sourceRoot = "${finalAttrs.src.name}/frontend";
+
+    npmDepsHash = "sha256-ImBNqccAd61c9ASzklcooQyh7BYdgJW9DTcQRmFHqho=";
+    makeCacheWritable = true;
+
+    preBuild = ''
+      chmod -R u+w ..
+    '';
+
+    installPhase = ''
+      runHook preInstall
+      cp -R .. $out
+      runHook postInstall
+    '';
+  });
+in
+python3Packages.buildPythonApplication {
   pname = "fava";
   version = "1.30.5";
   pyproject = true;
 
-  src = fetchPypi {
-    inherit pname version;
-    hash = "sha256-pfnRNhAcyuYFHqPBF0qCrK7w1PJiMOdYXCGj+xXi6uQ=";
-  };
+  inherit src;
+
+  patches = [ ./dont-compile-frontend.patch ];
 
   postPatch = ''
     substituteInPlace tests/test_cli.py \

